test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp

```diff
@@ -0,0 +1,436 @@
+    // The type is being used by another topic yet
+    EXPECT_NE(eprosima::fastdds::dds::TypeSupport(nullptr), statistics_participant->find_type(
+            count_type.get_type_name()));
+
+    ret = statistics_participant->disable_statistics_datawriter(DATA_COUNT_TOPIC);
+    ASSERT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
+    EXPECT_EQ(nullptr, statistics_participant->lookup_topicdescription(DATA_COUNT_TOPIC));
+    // The type is being used by another topic yet
+    EXPECT_NE(eprosima::fastdds::dds::TypeSupport(nullptr), statistics_participant->find_type(
+            count_type.get_type_name()));
+
+    ret = statistics_participant->disable_statistics_datawriter(PDP_PACKETS_TOPIC);
+    ASSERT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
+    EXPECT_EQ(nullptr, statistics_participant->lookup_topicdescription(PDP_PACKETS_TOPIC));
+    // The type is being used by another topic yet
+    EXPECT_NE(eprosima::fastdds::dds::TypeSupport(nullptr), statistics_participant->find_type(
+            count_type.get_type_name()));
+
+    ret = statistics_participant->disable_statistics_datawriter(EDP_PACKETS_TOPIC);
+    ASSERT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
+    EXPECT_EQ(nullptr, statistics_participant->lookup_topicdescription(EDP_PACKETS_TOPIC));
+    EXPECT_EQ(eprosima::fastdds::dds::TypeSupport(nullptr), statistics_participant->find_type(
+            count_type.get_type_name()));
+
+    ret = statistics_participant->disable_statistics_datawriter(DISCOVERY_TOPIC);
+    ASSERT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
+    EXPECT_EQ(nullptr, statistics_participant->lookup_topicdescription(DISCOVERY_TOPIC));
+    EXPECT_EQ(eprosima::fastdds::dds::TypeSupport(nullptr), statistics_participant->find_type(
+            discovery_type.get_type_name()));
+
+    ret = statistics_participant->disable_statistics_datawriter(SAMPLE_DATAS_TOPIC);
+    ASSERT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
+    EXPECT_EQ(nullptr, statistics_participant->lookup_topicdescription(SAMPLE_DATAS_TOPIC));
+    EXPECT_EQ(eprosima::fastdds::dds::TypeSupport(nullptr), statistics_participant->find_type(
+            sample_identity_count_type.get_type_name()));
+
+    ret = statistics_participant->disable_statistics_datawriter(PHYSICAL_DATA_TOPIC);
+    ASSERT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
+    EXPECT_EQ(nullptr, statistics_participant->lookup_topicdescription(PHYSICAL_DATA_TOPIC));
+    EXPECT_EQ(eprosima::fastdds::dds::TypeSupport(nullptr), statistics_participant->find_type(
+            physical_data_type.get_type_name()));
+
+    ASSERT_TRUE(eprosima::fastdds::dds::DomainParticipantFactory::get_instance()->
+            delete_participant(statistics_participant) == 
+            eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
+#endif // FASTDDS_STATISTICS
```

Comment:
Delete the standard participant in both cases.

```suggestion
#endif // FASTDDS_STATISTICS

    EXPECT_EQ(eprosima::fastdds::dds::DomainParticipantFactory::get_instance()->
            delete_participant(participant),
            eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
```

Reply:
Done in b295b5fd0a602a3075552ff51925ccfb8afa32b3